src/actions/graph.js

```diff
@@ -39,21 +79,46 @@ const chooseLayout = (instance, state) =>
 // Prepare new layout worker
 const layoutWorker = preventWorkerQueues(worker, chooseLayout);
 
+/**
+ * Formula to estimate the point at which the graph will take unreasonably
+ * long to render in terms of input nodes and edges.
+ *  @param {Object} largeGraphThreshold The defined threshold for a large graph
+ *  @param {integer} nodeCount The amount of nodes to be displayed in the flowchart
+ *  @param {integer} edgeCount The amount of edges to be displayed in the flowchart
+ */
+const isLarge = (largeGraphThreshold, nodeCount, edgeCount) => {
+  return nodeCount + 1.5 * edgeCount > largeGraphThreshold ? true : false;
+};
+
 /**
  * Async action to calculate graph layout in a web worker
  * whiled displaying a loading spinner
  * @param {Object} graphState A subset of main state
  * @return {function} A promise that resolves when the calcuation is done
  */
-export function calculateGraph(graphState) {
+export function calculateGraph(
+  graphState,
+  displayThreshold = largeGraphThreshold
+) {
   if (!graphState) {
     return updateGraph(graphState);
   }
   return async function(dispatch) {
-    dispatch(toggleLoading(true));
-    const graph = await layoutWorker(graphState);
-    dispatch(toggleGraph(true));
-    dispatch(toggleLoading(false));
-    return dispatch(updateGraph(graph));
+    const { nodes, edges, displayLargeGraph } = graphState;
+
+    if (
+      isLarge(displayThreshold, nodes.length, edges.length) === true &&
+      displayLargeGraph === false
+    ) {
+      dispatch(toggleIsLarge(true));
+      dispatch(updateNodeCount(nodes.length));
```

Comment:
Why dispatch an action to update the node count? You don't need an action to retrieve a value from the state. Why not use a selector instead? This action doesn't appear necessary.

Reply:
good call, not sure why I haven't thought of that 👍 

I've refactored to use the nodes selector and deleted the nodeCount field in the state.